types/src/shared/cache.ts

```diff
@@ -0,0 +1,48 @@
+import { redisClient } from "../shared/redis_client";
+
+// Wrapper function to cache the result of a function with Redis.
+// Usage:
+// const cachedFn = cacheWithRedis(fn, (fnArg1, fnArg2, ...) => `${fnArg1}-${fnArg2}`, 60 * 10 * 1000);
+export function cacheWithRedis<T extends (...args: any[]) => Promise<any>>(
```

Comment:
does it work if we replace the 2 `any` by `unknown` ?

Reply:
it won't. the caller sees a function with the same type as the one passed as a parameter. It's just that this function can take any type of function, so it has to be any.